halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesClouddriverBootstrapService.java

```diff
@@ -49,7 +50,7 @@ public Settings buildServiceSettings(DeploymentConfiguration deploymentConfigura
     List<String> profiles = new ArrayList<>();
     profiles.add("bootstrap");
     Settings settings = new Settings(profiles);
-    String location = "spinnaker";
+    String location = StringUtils.isEmpty(deploymentConfiguration.getDeploymentEnvironment().getLocation()) ? "spinnaker" : deploymentConfiguration.getDeploymentEnvironment().getLocation();
```

Comment:
I think we can avoid repeating this big ternary operation everywhere by dropping it into a function in DeploymentEnvironment.